zingolib/src/wallet/data.rs

```diff
@@ -967,8 +970,13 @@ impl TransactionMetadata {
         }
     }
 
-    pub fn get_transaction_fee(&self) -> u64 {
-        self.total_value_spent() - (self.value_outgoing() + self.total_change_returned())
+    pub fn get_transaction_fee(&self) -> Result<u64, ZingoLibError> {
```

Comment:
Hi @AloeareV I think this code is ready to merge, but since it's returning a `u64` it will interact with your NonNegativeAmount port.